app/src/main/java/nu/yona/app/ui/message/MessageStickyRecyclerAdapter.java

```diff
@@ -64,63 +71,137 @@ public MessageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
+				break;
+			default:
+				break;
 		}
+		return messagetitle;
+	}
+
+	private void setUpMessageItemHolderSwipe()
+	{
+		if (currentYonaMessage.getLinks() != null && currentYonaMessage.getLinks().getEdit() != null)
+		{
+			currentMessageItemHolder.swipeLayout.setRightSwipeEnabled(true);
+		}
+		else
+		{
+			currentMessageItemHolder.swipeLayout.setRightSwipeEnabled(false);
+		}
+	}
 
+	private void setUpMessageItemHolderDetails()
+	{
+		setUpMessageItemHolderSwipe();
+		if (currentYonaMessage.getNotificationMessageEnum() == NotificationMessageEnum.SYSTEM_MESSAGE)
+		{
+			setUpSystemMessageItemHolder();
+		}
+		else if (!TextUtils.isEmpty(currentYonaMessage.getNickname()))
+		{
+			setUpNonSystemMessageItemHolder();
+		}
+		setUpMessageContainerBackground();
+	}
+
+	private void setUpSystemMessageItemHolder()
+	{
+		currentMessageItemHolder.img_avtar.setVisibility(View.GONE);
+		if (!TextUtils.isEmpty(currentYonaMessage.getMessage()))
+		{
+			currentMessageItemHolder.txtFooterMsg.setText(currentYonaMessage.getMessage());
+		}
+		currentMessageItemHolder.profileIconTxt.setVisibility(View.VISIBLE);
+		currentMessageItemHolder.profileIconTxt.setText(currentYonaMessage.getNickname().substring(0, 1).toUpperCase());
+		currentMessageItemHolder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_admin_round));
+	}
+
+	private void setUpNonSystemMessageItemHolder()
+	{
+		currentMessageItemHolder.txtFooterMsg.setText(currentYonaMessage.getNickname());
+		if (currentYonaMessage.getNotificationMessageEnum() == NotificationMessageEnum.GOALCONFLICTMESSAGE_ANNOUNCED)
+		{
+			currentMessageItemHolder.img_avtar.setImageResource(R.drawable.adult_sad);
+			currentMessageItemHolder.img_avtar.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.GONE);
+		}
+		else
+		{
+			setMessageListItemUserAvatar();
+		}
+	}
+
+	private void setMessageListItemUserAvatar()
+	{
+		if (currentYonaMessage.getLinks().getUserPhoto() != null)
+		{
+			Picasso.with(this.activity).load(currentYonaMessage.getLinks().getUserPhoto().getHref()).noFade().into(currentMessageItemHolder.img_avtar);
+			currentMessageItemHolder.img_avtar.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.GONE);
+		}
+		else
+		{
+			currentMessageItemHolder.img_avtar.setVisibility(View.GONE);
+			currentMessageItemHolder.profileIconTxt.setVisibility(View.VISIBLE);
+			currentMessageItemHolder.profileIconTxt.setText(currentYonaMessage.getNickname().substring(0, 1).toUpperCase());
+			currentMessageItemHolder.profileIconTxt.setBackground(ContextCompat.getDrawable(YonaActivity.getActivity(), R.drawable.bg_small_self_round));
+		}
+	}
+
+	private void setUpMessageContainerBackground()
+	{
+		if (currentYonaMessage.getLinks() != null && currentYonaMessage.getLinks().getMarkRead() != null && !TextUtils.isEmpty(currentYonaMessage.getLinks().getMarkRead().getHref()))
```

Comment:
This could be made quite a bit more readable (no need to check whether ``getHref`` returns non-null):
```
boolean isUnread = currentYonaMessage.getLinks() != null && currentYonaMessage.getLinks().getMarkRead() != null;
int resourceId = (isUnread) ? R.drawable.item_selected_gradient : R.drawable.item_gradient;
currentMessageItemHolder.messageContainer.setBackground(ContextCompat.getDrawable(activity, resourceId));
```

Reply:
Done.